# КОММЕНТАРИЙ РЕВЬЮВЕРА
Привет 👋!)

Меня зовут Сибирев Владислав, я осуществлял проверку проекта. Может перейдем сразу на <b>ты</b>? Если это неприемлемо, то обязательно дай знать)

Далее по ходу работы ты сможешь встретить мои комментарии, они будут отмечены разными цветами:

<div class="alert alert-block alert-success">
 ✔️ Похвала и возможности развития
</div>
<div class="alert alert-block alert-warning">
⚠️ Некритичные замечания, но стоит обратить внимание
</div>
<div class="alert alert-block alert-danger">
❌ Грубая ошибка, необходимо исправить
</div>


Заранее прошу не удалять мои комментарии, это важно. Если у тебя возникнут вопросы, обозначь их, я с радостью тебе помогу!)

Как вариант, можешь обозначить их так:
<div class="alert alert-block alert-info">
<b>КОММЕНТАРИЙ СТУДЕНТА:</b>
</div>

# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
# Создаем сводную таблицу, в которой группируем данные в зависимости от количества детей у Заемщика
data_pivot_children = data.pivot_table(index = ['children'], values = 'debt', aggfunc = ['sum','count','mean'])

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера:</b>

✔️ Отличная работа в части вывода информации, молодец, что вывел долю должников с помощью функции `mean`))

In [23]:
# Переименование столбцов 
data_pivot_children.columns = ['Кол-во должников','Кол-во заемщиков', 'Процент должников'] 

# Отдельно переименовываем столбец с индексом, значения в котором совпадает с количеством детей
data_pivot_children.index.name = 'Кол-во детей' 


In [24]:
# Округление значения столбца 'Процент должников'
data_pivot_children['Процент должников'] = (data_pivot_children['Процент должников']*100).round(1)

# Приводим значение столбца 'Процент должников' к строчному типу
data_pivot_children['Процент должников']= data_pivot_children['Процент должников'].astype('str') 

# Добавление символа '%' к значениям столбца 'Процент должников'
data_pivot_children['Процент должников']= data_pivot_children['Процент должников'] + '%'

# Вывод таблицы
data_pivot_children 

Кол-во должников  Кол-во заемщиков Процент должников
Кол-во детей                                                      
0                         1063             14091              7.5%
1                          444              4808              9.2%
2                          194              2052              9.5%
3                           27               330              8.2%
4                            4                41              9.8%
5                            0                 9              0.0%

В получившихся данных, в связи с очень маленькой выборкой, мы не можем корректно сравнить категории Заемщиков, имеющих трех и более детей. По оставшимся категориям видно, что Заемщики не имеющие детей, лучше всего отдают кредиты, их доля составляет 7,5%. Заемщики, имеющие одного или двух детей, намного чаще не возвращают кредиты вовремя, их доля составляет 9,2% и 9,5% соответственно. Полученные данные указывают на наличие зависимости между количеством детей и возвратом кредита в срок.

<div class="alert alert-block alert-warning">
<b>Комментарий ревьюер:</b> 

⚠️ Молодец, что используешь в выводах цифры, также нужно сделать акцент на сбалансированности выборок. Примерно выводы будут выглядеть так: 
`люди, которые не имеют детей, лучше всего отдают кредит, доля должников здесь составляет с 7,5%, хуже обстоит дело с должниками, кто имеет одного или двух детей, здесь доля невозврата составляет соответственно 9,2% и 9,4%. Остальные категории уже некорректно сравнивать, так как в них очень маленькая выборка`)) <br>

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера_V2:</b>

✔️  Выводы переработал, молодец)) 

  </div>

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [25]:
#Создаем сводную таблицу, в которой группируем данные в зависимости от семейного положения Заемщика
data_pivot_family = data.pivot_table(index = ['family_status'], values = 'debt', aggfunc = ['sum','count','mean'])

In [26]:
#Сбрасываем индекс в получившейся таблице
data_pivot_family = data_pivot_family.reset_index()

#Переименовываем стобцы таблицы
data_pivot_family.columns = ['Семейное положение','Кол-во должников','Кол-во заемщиков', 'Процент должников']


In [27]:
# Округление значения столбца 'Процент должников'
data_pivot_family['Процент должников'] = (data_pivot_family['Процент должников']*100).round(1)

# Приводим значение столбца 'Процент должников' к строчному типу
data_pivot_family['Процент должников']= data_pivot_family['Процент должников'].astype('str')

# Добавление символа '%' к значениям столбца 'Процент должников'
data_pivot_family['Процент должников']= data_pivot_family['Процент должников'] + '%'


# вывод таблицы
data_pivot_family


Семейное положение  Кол-во должников  Кол-во заемщиков Процент должников
0  Не женат / не замужем               273              2796              9.8%
1              в разводе                84              1189              7.1%
2         вдовец / вдова                63               951              6.6%
3       гражданский брак               385              4134              9.3%
4        женат / замужем               927             12261              7.6%

В получившихся данных, в связи с маленькой выборкой, мы не можем корректно сравнить Заемщиков из категории 'в разводе' и 'вдовец/вдова'. Лучшие показатели по возврату кредита у Заемщиков из категории 'женат/замужем', их доля 7,6%. Показатели возврата кредита в Категориях 'гражданский брак' и 'не женат/не замужем' составляют 9,3% и 9,8% соответственно. Полученные данные указывают на наличие зависимости между семейным положением и возвратом кредита в срок.

<div class="alert alert-block alert-warning">
<b>Комментарий ревьюер:</b> 

⚠️ Удели здесь внимание тоже сбалансированности выборок, по большому счету здесь можно сравнивать только три группы)) 

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера_V2:</b>

✔️  Здесь тоже все отлично по выводам))

  </div>

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [28]:
#Создаем сводную таблицу, в которой группируем данные в зависимости от уровня дохода Заемщика
data_pivot_total_income_category = data.pivot_table(index = ['total_income_category'], values = 'debt', aggfunc = ['sum','count','mean'])

In [29]:
# Сбрасываем индекс в получившейся таблице
data_pivot_total_income_category = data_pivot_total_income_category.reset_index()

# Переименовываем стобцы таблицы
data_pivot_total_income_category.columns = ['Уровень дохода','Кол-во должников','Кол-во заемщиков', 'Процент должников']


In [30]:
# Округление значения столбца 'Процент должников'
data_pivot_total_income_category['Процент должников'] = (data_pivot_total_income_category['Процент должников']*100).round(1)

# Приводим значение столбца 'Процент должников' к строчному типу
data_pivot_total_income_category['Процент должников']= data_pivot_total_income_category['Процент должников'].astype('str')

# Добавление символа '%' к значениям столбца 'Процент должников'
data_pivot_total_income_category['Процент должников']= data_pivot_total_income_category['Процент должников'] + '%'

# Вывод таблицы
data_pivot_total_income_category

Уровень дохода  Кол-во должников  Кол-во заемщиков Процент должников
0              A                 2                25              8.0%
1              B               354              5014              7.1%
2              C              1353             15921              8.5%
3              D                21               349              6.0%
4              E                 2                22              9.1%

В получившихся данных, в связи с маленькой выборкой, мы не можем корректно сравнить Заемщиков из категории 'А', 'D' и 'Е'. В категориях 'B' и 'C' доля возврата кредита составляет 7,1% и 8,5%, что говорить о зависимости между уровнем дохода и возвратом кредита в срок.

<div class="alert alert-block alert-warning">
<b>Комментарий ревьюер:</b> 

⚠️ Здесь сравниваем только две группы B и С, не забываем про сбалансированость выборки))<br>

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера_V2:</b>

✔️  Здесь сравнил две самые большие группы))

  </div>

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [31]:
# Создаем сводную таблицу, в которой группируем данные в зависимости от целей кредита
data_pivot_purpose_category = data.pivot_table(index = ['purpose_category'], values = 'debt', aggfunc = ['sum','count','mean'])

In [32]:
# Сбрасываем индекс в получившейся таблице
data_pivot_purpose_category = data_pivot_purpose_category.reset_index()

# Переименовываем стобцы таблицы
data_pivot_purpose_category.columns = ['Цели кредита','Кол-во должников','Кол-во заемщиков', 'Процент должников']

In [33]:
# Округление значения столбца 'Процент должников'
data_pivot_purpose_category['Процент должников'] = (data_pivot_purpose_category['Процент должников']*100).round(1)

# Приводим значение столбца 'Процент должников' к строчному типу
data_pivot_purpose_category['Процент должников']= data_pivot_purpose_category['Процент должников'].astype('str')

# Добавление символа '%' к значениям столбца 'Процент должников'
data_pivot_purpose_category['Процент должников']= data_pivot_purpose_category['Процент должников'] + '%'

# Вывод таблицы
data_pivot_purpose_category

Цели кредита  Кол-во должников  Кол-во заемщиков  \
0    операции с автомобилем               400              4279   
1  операции с недвижимостью               780             10751   
2     получение образования               369              3988   
3        проведение свадьбы               183              2313   

  Процент должников  
0              9.3%  
1              7.3%  
2              9.3%  
3              7.9%

Чаще всего в срок не возвращают кредиты, взятые на покупку автомобиля и на получение образования - 9.3%. Процент невозврата кредитов, взятых на покупку недвижимости и проведение свадьбы составляет - 7,3% и 7,9% соответственно. Самыми надежными Заемщики являются те, кто берет ипотеку. Вероятнее всего это связанно с тем, что изначально Заемщикам из более рискованных категорий реже одобряют ипотечные кредиты, т.к. в связи со значительностью суммы ипотечного кредита для его получения необходимо соблюдение более жестких требований (возраст, доход и т.д.).  

<div class="alert alert-block alert-warning">
<b>Комментарий ревьюера:</b> 

⚠️ Почему по твоему самые надежные заемщики те, кто берет ипотеку? Здесь можно порасуждать в выводах)) 

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера_V2:</b>

✔️ Все верно, самые большие суммы при ипотеке, поэтому банки предъявляют самые жесткие требования к ипотечникам, поэтому их проверяют вдоль и поперек при выдаче кредитов, плюс собственность находится в залоге у банка, не будешь платить она вернется к банку)) 
   
       
  </div>

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

Пропуски в исходных данных могут появляться из-за человеческого фактора, технической ошибки, распознование текста, объединения или разделения данных

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера:</b>

✔️ Ответ правильный)) 
  </div>

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

Медианные значения более корректно отражают данные, в которых есть значения, сильно выделяющиеся на фоне других. Метод позволяет получить более объективную информацию о имеющихся данных.

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера:</b>

✔️ Медиана не подвержена выбросам, все верно)) 

### Шаг 4: общий вывод.

Был проведен анализ статистики платёжеспособности клиентов с целью разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. В результате проведенного анализ получены следующие выводы:

**Вывод. Дети**

В результате анализа предоставленных данных, можно утверждать, что имеется зависимость возврата кредита в срок от количества детей у Заемщика. Лучше всего отдают кредиты Заемщики не имеющие детей, их доля составляет 7,5%. Доля Заемщиков, имеющие одного или двух детей, составляет 9,2% и 9,5% соответственно, т.е. доля невозврата кредита растет с увеличением количества детей у Заемщика

**Вывод. Семейное положение**

По итогам анализа данных, наблюдается зависимость возврата кредита в срок от семейного положения Заемщика. Лучшие доля по возврату кредита у Заемщиков из категории 'женат/замужем' - 7,6%. Доля в категориях 'гражданский брак' и 'не женат/не замужем' составляют 9,3% и 9,8% соответственно, т.е. заемщики, состоящие в официальных семейных отношениях, чаще возвращают кредит вовремя.

**Вывод. Уровень дохода**

В связи с маленькой выборкой, корректно можно сравнить данных только из двух категорий 'B' и 'С' ('B' - Доход выше среднего, 'С' - Средний доход). В категориях 'B' и 'C' доля возврата кредита составляет 7,1% и 8,5%, т.е. чем выше доход, тем чаще возвращается кредит.

**Вывод. Цели кредита**

По итогам сравнения целей кредита, наихудшие показатели у кредитов, взятых на покупку автомобиля и на получение образования - 9.3%. Лучшие показатели у ипотечных кредитов - 7,3%. Доля кредитов, взятых на проведение свадьбы - 7,9%. Такие показатели ипотечных кредитов связанны с тем, что изначально из-за значительности суммы ипотечного кредита для его получения необходимо соблюдение более жестких требований (возраст, доход и т.д.).  

**Итог**

Из проведенного анализа можно сделать вывод, что количество детей, семейное положение, доход и цели кредита влияют на его возврат. Наилучшие показатели у Заемщиков с доходом выше среднего, в браке и не имеющие детей, берущих ипотечный кредит. Наиболее рискованные Заемщики, не состоящие в семейных отношениях, имеющие одного и более детей, со средним доходом и ниже, берущих кредит на покупку автомобиля или образование.

 <div class="alert alert-block alert-success">
<b>Комментарий ревьюера_V2:</b>

✔️ Выводы отлично переработал, молодец)) Рекомендации даны банку, исследование проведено))
  
  
</div>

<div class="alert alert-block alert-warning">
<b>Комментарий ревьюера:</b> 

⚠️ По структуре все отлично, но общий вывод - резюме этой работы. Большинство заказчиков посмотрят сначала его (а некоторые только его). Чтобы выводы не огорошивали заказчика своей очевидностью, стоит добавить в них цифр, всяких корреляций, процентов итд :) Сделай так, чтобы вся найденная в данных информация была отражена в выводе))<br> Но одним текстом тяжело читается выводы, надо разбивать все по абзацам, вот так вот:
- Вывод про детей
- Вывод про семейное положение
- Так далее

    
    
Я дам тебе ссылки на полезные гайды, которые могут помочь:
    
<a href="https://markdown-guide.readthedocs.io/en/latest/basics.html">guide1.</a>
<a href="https://paulradzkov.com/2014/markdown_cheatsheet/">guide2.</a>

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера:</b>
    
Дмитрий, отлично поработал, видно что умеешь работать с кодом, в общем молодец, но нужно будет поработать над выводами)) 
<br> Жду исправленного варианта)
  </div>

<div class="alert alert-block alert-info">
<b>КОММЕНТАРИЙ СТУДЕНТА:</b>

Владислав, приветствую! Спасибо за комментарии, постарался всё учесть, особенно про сбалансированность выборки, очень помогло. Правильно делать выводы тоже нужно учиться) 
Направляю новый вариант. Всегда готов выслушать новые замечания и комментарии. 
</div>

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера:</b>
    
Дмитрий, привет) Спасибо большое за обратную связь) Выводы это тоже очень важный навык аналитика, так как аналитик не просто пишет код и строит графики, а все это анализирует, подмечая основные моменты в исследовании, на которые стоит обратить)) Отличная работа, можно двигаться дальше)) Успехов в дальнейшем обучении)) 
  </div>